src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts

```diff
@@ -452,6 +471,84 @@ export class InstructorSessionEditPageComponent extends InstructorSessionBasePag
     });
   }
 
+  /**
+   * Prompts the user to delete individual extensions that are before or equal to the new session end time.
+   */
+  deleteDeadlineExtensionsHandler(submissionEndTimestamp: number): Observable<boolean> {
+    const [studentDeadlinesToDelete, instructorDeadlinesToDelete] = this
+      .getIndividualDeadlinesToDelete(submissionEndTimestamp);
+
+    const isAllDeadlinesAfterUpdatedEndTime = studentDeadlinesToDelete.length === 0
+      && instructorDeadlinesToDelete.length === 0;
+
+    if (isAllDeadlinesAfterUpdatedEndTime) {
+      return of(true); // no need to prompt for deletion
+    }
+
+    const [affectedStudentModels, affectedInstructorModels] = this
+      .getAffectedIndividualModels(submissionEndTimestamp, studentDeadlinesToDelete, instructorDeadlinesToDelete);
+
+    const modalRef: NgbModalRef = this.ngbModal.open(ExtensionConfirmModalComponent);
+    modalRef.componentInstance.modalType = ExtensionModalType.SESSION_DELETE;
+    modalRef.componentInstance.selectedStudents = affectedStudentModels;
+    modalRef.componentInstance.selectedInstructors = affectedInstructorModels;
+    modalRef.componentInstance.extensionTimestamp = submissionEndTimestamp;
+    modalRef.componentInstance.feedbackSessionTimeZone = this.sessionEditFormModel.timeZone;
+
+    return new Observable((subscribeIsUserAccept) => {
+      modalRef.componentInstance.onConfirmExtensionCallBack.subscribe(() => {
+        this.removeDeadlines(affectedStudentModels, affectedInstructorModels);
+        modalRef.componentInstance.isSubmitting = false;
+        modalRef.close();
+        subscribeIsUserAccept.next(true);
+      }, () => {
```

Comment:
Since you are relying on `onConfirmExtensionCallBack`, would it not be possible to use `pipe` and `tap` here instead of creating a new `Observable`?

Reply:
It's possible, but the level of abstraction won't quite match if I were to call `updateFeedbackSession` in the pipe

Reply:
As mentioned [here](https://github.com/TEAMMATES/teammates/pull/11704#discussion_r851595511), I did not mean for `updateFeedbackSession` to be piped/tapped here too, just whatever is already inside there. But similar to the other comment, this may not be completely necessary so it's fine to just leave it like this I think.